src/three/plugins/images/sources/GeoJSONImageSource.js

```diff
@@ -0,0 +1,386 @@
+		fillStyle = 'rgba( 255,0,0,0.65 )',
+	} = {} ) {
+
+		super();
+		this.geojson = geojson;
+		this.url = url;
+		this.tileDimension = tileDimension;
+		this.levels = levels;
+
+		this.pointRadius = pointRadius;
+		this.strokeStyle = strokeStyle;
+		this.strokeWidth = strokeWidth;
+		this.fillStyle = fillStyle;
+
+	}
+
+	async init() {
+
+		// configure tiling to content bounds and levels
+		const { tiling, levels, tileDimension, geojson, url } = this;
+		const projection = new ProjectionScheme();
+		tiling.setProjection( projection );
+		tiling.setContentBounds( ...projection.getBounds() );
+
+		tiling.generateLevels(
+			levels,
+			projection.tileCountX,
+			projection.tileCountY,
+			{
+				tilePixelWidth: tileDimension,
+				tilePixelHeight: tileDimension,
+			},
+		);
+
+		// If a URL was provided and no geojson object yet, fetch it now (use fetchData so overlay can inject headers)
+		if ( ! geojson && url ) {
+
+			const res = await this.fetchData( url );
+			this.geojson = await res.json();
+
+		}
+
+		// TODO: If geojson present, compute bounds from data (with padding) and set as content bounds.
+		// Falls back to full projection bounds if no geojson or unable to compute an extent.
+
+		// seems that this approach to optimize rendering by defining bounds
+		// doesn't work if not using a very large offset for the bounds, maybe an error?
+		// const geoBounds = this._geoJSONBounds( 50 );
+		// if ( geoBounds ) {
+		// 	this.tiling.setContentBounds( ...geoBounds );
+		// 	console.log( `GeoJSONImageSource: set content bounds from geojson: ${geoBounds}` );
+		// } else {
+		// 	this.tiling.setContentBounds( ...this.tiling.projection.getBounds() );
+		// }
+
+	}
+
+	// main fetch per tile - > returns .Texture
+	async fetchItem( tokens, signal ) {
+
+		return this.drawCanvasImage( tokens );
+
+	}
+
+	drawCanvasImage( tokens ) {
+
+		const { tiling, tileDimension, geojson } = this;
+		const [ x, y, level ] = tokens;
+		const tileBoundsDeg = tiling.getTileBounds( x, y, level, false, false ).map( v => MathUtils.RAD2DEG * v );
+
+		// create canvas
+		const canvas = document.createElement( 'canvas' );
+		canvas.width = tileDimension;
+		canvas.height = tileDimension;
+
+		// draw features
+		const ctx = canvas.getContext( '2d' );
+		const features = this._featuresFromGeoJSON( geojson );
+		for ( let i = 0; i < features.length; i ++ ) {
+
+			const feature = features[ i ];
+			if ( this._featureIntersectsTile( feature, tileBoundsDeg ) ) {
+
+				this._drawFeatureOnCanvas( ctx, feature, tileBoundsDeg, canvas.width, canvas.height );
+
+			}
+
+		}
+
+		const tex = new CanvasTexture( canvas );
+		tex.needsUpdate = true;
+		return tex;
+
+	}
+
+	// Compute pixel from projected coords and tileInfo
+	_projectedToPixel( pxProjX, pxProjY, bounds, width, height ) {
+
+		let [ minX, minY, maxX, maxY ] = bounds;
+
+		// avoid zero-range (prevents division by zero in mapLinear)
+		if ( maxX === minX ) maxX = minX + 1;
+		if ( maxY === minY ) maxY = minY + 1;
```

Comment:
Likewise - can you explain this case? In order for this case to be encountered the "tile bounds" would have to have a total width of "0" in either dimension, right? I don't think it's possible for that to happen?